woocommerce/google-listings-and-ads
# [API Pull] Auth Flow

### Changes proposed in this Pull Request:

- Auth flow for granting access to WPCOM after enabling the product sync feature
- Tweak the banner
- Revoke access when disabling the feature

### Detailed test instructions:


1. Go to Settings and verify you see "We will soon transition to a new and improved method for synchronizing product data with Google -- Get early access" banner 
2. Click on Get early access (see the button loading)
3. You will be redirected to auth flow
4. Complete the auth flow
5. You will be redirected back to the store
6. You see "Google has been granted access to fetch your product data." in green on the Google MC card 
7. Click on "Disable product data fetch"
8. Auth is disabled and you see the banner again


### Additional details:

- Deeper unit Tests coming on further PRs
- Consistency and more tracking events coming in further PRs
- Notice that the errors are still not handled from Google. You can see that for example removing the nonce param in `get_data_from_google` function
